### PR TITLE
Footer

### PR DESCRIPTION
**Pull request by drerchi**

### CODE REVIEW

File: src/components/footer/ChatLocation.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+`;
+
+const ChatLocation: FC<RouteComponentProps> = () => (
+  <LocationDiv>
+    <InlineDiv>
+      <Place />
+    </InlineDiv>{" "}
+    <Lato>
+      Strojárenská 3,
+      <br />
```
Comment (Stephaaan) [inline src/components/footer/ChatLocation.tsx]: Don't use `<br/>` tags
fix everywhere

File: src/components/icons/Cogs.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React, { FunctionComponent } from "react";
```
Comment (garySiladi) [inline src/components/icons/Cogs.tsx]: I would change the file structure of these /icons/something.tsx files. Seems to me very repetitive, and contains the same div with img in it. I would create a universal icon component which receives a style, and src prop.

File: src/components/footer/ChatLogo.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+import test from "../static/svg/icon-power-off.svg";
+
+const GilbertBold = styled.div`
+  font-family: "Gilbert", serif;
+  font-size: 50px;
+  font-weight: bold;
+  font-style: normal;
+  font-color: #333333;
```
Comment (Stephaaan) [inline src/components/footer/ChatLogo.tsx]: font color hardcoded 
in future - use value from colors.ts file

File: src/components/footer/FooterMobile.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+const FooterDiv = styled.div`
+  height: 270px;
+  width: 375px;
+  background-color: #ffc450;
+  padding-top: 200px;
+`;
+
+const Footer: FC<RouteComponentProps> = () => (
+  <div>
```
Comment (garySiladi) [inline src/components/footer/FooterMobile.tsx]: why the div?

File: src/components/footer/FooterDesktop.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  padding-left: 44px;
+`;
+
+const FooterInfoHandler = styled.div`
+  display: flex;
+`;
+
+const Footer: FC<RouteComponentProps> = () => (
+  <FooterDiv>
+    <div style={{ width: "1024px" }}>
```
Comment (garySiladi) [inline src/components/footer/FooterDesktop.tsx]: this div isn't needed, as the parent has the same width

File: src/components/footer/FooterDesktop.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import React, { FC } from "react";
```
Comment (garySiladi) [inline src/components/footer/FooterDesktop.tsx]: we need one component for mobile and desktop, take care of differences in css.

File: src/components/footer/ChatLogo.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+import test from "../static/svg/icon-power-off.svg";
+
+const GilbertBold = styled.div`
```
Comment (garySiladi) [inline src/components/footer/ChatLogo.tsx]: name this after the font but purpose. like `LogoHeader`

File: src/components/footer/ChatLogo.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+import test from "../static/svg/icon-power-off.svg";
```
Comment (garySiladi) [inline src/components/footer/ChatLogo.tsx]: what is this?

File: src/components/footer/ChatLogo.tsx (hunk 3 above)
Comment (garySiladi) [inline src/components/footer/ChatLogo.tsx]: use const colors

File: src/components/footer/ChatICO.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
```
Comment (Stephaaan) [inline src/components/footer/ChatICO.tsx]: don't use RouteComponentProps, make own interface
use RouteComponentProps just in a components which are children of Router component from @reach/router

File: src/components/footer/ChatLocation.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+import Place from "../icons/Place";
+
+const LocationDiv = styled.div`
+  width: 145px;
```
Comment (garySiladi) [inline src/components/footer/ChatLocation.tsx]: could have em or percentages rather than px.

File: src/components/footer/ChatLocation.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+  font-style: normal;
+  font-color: black;
+  display: inline-block;
+`;
+
+const ChatLocation: FC<RouteComponentProps> = () => (
+  <LocationDiv>
+    <InlineDiv>
+      <Place />
+    </InlineDiv>{" "}
```
Comment (garySiladi) [inline src/components/footer/ChatLocation.tsx]: {" "}  isn't needed. solve with css

File: src/components/footer/ChatICO.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+`;
+
+const ChatInfo = styled.div`
+  width: 242px;
+  height: 36px;
+  display: inline-block;
+  padding-left: 24px;
+`;
+
+const Lato = styled.div`
```
Comment (Stephaaan) [inline src/components/footer/ChatICO.tsx]: repeating component code
what about creating a component once and importing it?

File: src/components/footer/ChatContacts.tsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+import Phone from "../icons/Phone";
+import Email from "../icons/Email";
+
+const InlineDiv = styled.div`
```
Comment (Stephaaan) [inline src/components/footer/ChatContacts.tsx]: use more meaningful name without element name in it

File: src/components/footer/ChatLocation.tsx [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+`;
+
+const InlineDiv = styled.div`
+  display: inline-block;
+`;
+
+const Lato = styled.div`
+  font-size: 12px;
+  font-family: Lato;
+  font-weight: normal;
```
Comment (garySiladi) [inline src/components/footer/ChatLocation.tsx]: isn't these `normal` values redundant, as inherited from parents?

File: src/components/footer/ChatICO.tsx [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+  font-weight: normal;
+  font-style: normal;
+  font-color: black;
+  display: inline-block;
+`;
+
+const ChatICO: FC<RouteComponentProps> = () => (
+  <div>
+    <ChatInfo>
+      <LatoBold>OZ CHAT - Centrum pre kreatívnu liečbu Arteterapiou</LatoBold>
```
Comment (garySiladi) [inline src/components/footer/ChatICO.tsx]: don't name tag after font name

**Review by garySiladi — CHANGES_REQUESTED** [review]

File: src/components/footer/FooterMobile.tsx [hunk 16]
```diff
@@ -0,0 +1,29 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+import ChatLogo from "./ChatLogo";
+import ChatICO from "./ChatICO";
+import ChatLocation from "./ChatContacts";
```
Comment (Stephaaan) [inline src/components/footer/FooterMobile.tsx]: importing chatlocation from chatcontacts???

File: src/components/footer/FooterMobile.tsx [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+import ChatContacts from "./ChatLocation";
```
Comment (Stephaaan) [inline src/components/footer/FooterMobile.tsx]: importing chatcontacts from chatlocation???

File: src/components/icons/Cogs.tsx (hunk 2 above)
Comment (Stephaaan) [inline src/components/icons/Cogs.tsx]: use shorthand version of import of FunctionCode (FC) everywhere

File: src/components/icons/Email.tsx [hunk 18]
```diff
@@ -0,0 +1,15 @@
+import React, { FunctionComponent } from "react";
+import styled from "@emotion/styled";
+import email from "../../static/svg/email.svg";
+
+const Email: FunctionComponent = () => (
+  <span>
+    <img
+      src={email}
+      alt="svg"
+      style={{ paddingBottom: "-18px", paddingRight: "8px" }}
```
Comment (Stephaaan) [inline src/components/icons/Email.tsx]: use custom styled component for positioning the image (if it's possible)

**Review by Stephaaan — CHANGES_REQUESTED** [review]
Please fix the things written in rewiew

File: src/components/footer/ChatContacts.tsx [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+  display: inline-block;
+`;
+
+const ContactsContainer = styled.div`
```
Comment (pggl) [inline src/components/footer/ChatContacts.tsx]: you don't need to write "Contacts" It's ChatContact component and the component is used only here so container for naming is good

File: src/components/footer/ChatContacts.tsx [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+`;
+
+const Contacts = styled.div`
+  font-family: Lato;
+  font-size: 12px;
+  font-color: black;
+  display: inline-block;
+`;
+
+const ChatContacts: FC<RouteComponentProps> = () => (
```
Comment (pggl) [inline src/components/footer/ChatContacts.tsx]: RouteComponentProps is necessary only for components which are used for Routing

File: src/components/footer/ChatICO.tsx [hunk 21]
```diff
@@ -0,0 +1,48 @@
+import React, { FC } from "react";
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+
+const LatoBold = styled.div`
```
Comment (pggl) [inline src/components/footer/ChatICO.tsx]: this type of components is reusable, don't write it again and again

File: src/components/footer/ChatLogo.tsx [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+`;
+
+const ChatContainer = styled.div`
+  width: 241px;
+  height: 91px;
+  padding-top: 30px;
+`;
+
+const ChatLogo: FC<RouteComponentProps> = () => (
+  <div>
```
Comment (pggl) [inline src/components/footer/ChatLogo.tsx]: unnecessary div

File: src/components/footer/FooterDesktop.tsx [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+import styled from "@emotion/styled";
+import { RouteComponentProps } from "@reach/router";
+import ChatLogo from "./ChatLogo";
+import ChatICO from "./ChatICO";
+import ChatLocation from "./ChatContacts";
+import ChatContacts from "./ChatLocation";
+
+const FooterDiv = styled.div`
```
Comment (pggl) [inline src/components/footer/FooterDesktop.tsx]: I think better name is wrapper or container

File: src/components/icons/Cogs.tsx [hunk 24]
```diff
@@ -0,0 +1,11 @@
...
+import styled from "@emotion/styled";
+import cogs from "../../static/svg/icon-cogs.svg";
+
+const Cogs: FunctionComponent = () => (
+  <div>
```
Comment (pggl) [inline src/components/icons/Cogs.tsx]: unnecessary div

File: src/components/icons/FullCheckbox.tsx [hunk 25]
```diff
@@ -0,0 +1,11 @@
+import React, { FunctionComponent } from "react";
+import styled from "@emotion/styled";
```
Comment (pggl) [inline src/components/icons/FullCheckbox.tsx]: why is here this import?

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/Router.tsx [hunk 26]
```diff
@@ -4,6 +4,7 @@ import Home from "./components/Home";
 import Example from "./components/Example";
 import Counter from "./components/Counter";
 import Testing from "./components/Testing";
+import Footer from "./components/footer/Footer";
```
Comment (garySiladi) [inline src/Router.tsx]: name the component `Footer.js` to `index.js` in the `footer` folder. That way the import will be simpler `"./components/footer"`

File: src/Router.tsx [hunk 27]
```diff
@@ -12,6 +13,7 @@ export default () => {
       <Example path="foo" />
       <Counter path="counter" />
       <Testing path="testing" />
+      <Footer path="footer" />
```
Comment (pggl) [inline src/Router.tsx]: remove this it doesn't have to be in the final code

Comment (pggl) [reply src/Router.tsx]: and also remove import

File: src/components/Button.tsx [hunk 28]
```diff
@@ -0,0 +1,34 @@
...
+
+const Text = styled.span`
+  font-family: Lato, serif;
+  font-size: 16px;
+  font-weight: 900;
+  font-style: normal;
+  color: ${colors.white};
+`;
+
+const Button: FC<IProps> = ({ text, backgroundColor = "black" }) => (
```
Comment (pggl) [inline src/components/Button.tsx]: why is there this component ?

File: src/components/Icon.tsx [hunk 29]
```diff
@@ -0,0 +1,10 @@
+import React, { FC, ReactNode } from "react";
+interface IProps {
+  src: string;
+  className?: string;
+  children?: ReactNode;
```
Comment (pggl) [inline src/components/Icon.tsx]: you don't need children it is already in FC

File: src/components/footer/Contacts.tsx [hunk 30]
```diff
@@ -0,0 +1,23 @@
...
+      <div className="mr-3 w-4">
+        <Icon src={Phone} />
+      </div>
+      <div className="font-lato text-xs color-black">+421 905 271 220</div>
+    </div>
+    <div className="pb-1 flex justify-start items-center">
+      <div className="mr-3 w-4">
+        <Icon src={Email} />
+      </div>
+      <div className="font-lato text-xs color-black">chatkosice@gmail.com</div>
```
Comment (pggl) [inline src/components/footer/Contacts.tsx]: if there is email make proper html

File: src/components/footer/Location.tsx [hunk 31]
```diff
@@ -0,0 +1,26 @@
...
+import styled from "@emotion/styled";
+import Icon from "../Icon";
+import Place from "../../static/svg/icon-place.svg";
+
+const Street = styled.div`
+  width: 172px;
+  height: 36px;
+`;
+
+const ChatLocation = () => (
```
Comment (pggl) [inline src/components/footer/Location.tsx]: file name is Location and component is ChatLocation I think this is not ok

File: src/components/footer/Logo.tsx [hunk 32]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+const ChatLogo = () => (
```
Comment (pggl) [inline src/components/footer/Logo.tsx]: filename and component name is not same

File: src/components/footer/index.tsx [hunk 33]
```diff
@@ -0,0 +1,23 @@
+import React, { FC } from "react";
+import { RouteComponentProps } from "@reach/router";
+import Logo from "./Logo";
+import ICO from "./Info";
+import Contacts from "./Contacts";
+import Location from "./Location";
+
+const Footer: FC<RouteComponentProps> = () => (
```
Comment (pggl) [inline src/components/footer/index.tsx]: remove RouteComponentProps

File: src/main.css [hunk 34]
```diff
@@ -0,0 +1,138 @@
+html,
```
Comment (pggl) [inline src/main.css]: remove this file, reset css is already in code

File: src/stories/index.tsx [hunk 35]
```diff
@@ -3,5 +3,7 @@ import { storiesOf } from "@storybook/react";
 import Example from "../components/Example";
 import "../style/index.css";
 import "../style/fonts.css";
+import Footer from "../components/footer";
 
 storiesOf("Example", module).add("example", () => <Example />);
+storiesOf("Footer", module).add("responsive footer", () => <Footer />);
```
Comment (pggl) [inline src/stories/index.tsx:9]: there should be two versions, desktop and mobile

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/components/footer/Info.tsx [hunk 36]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import styled from "@emotion/styled";
+
+const Street = styled.div`
+  width: 270px;
+  height: 36px;
+`;
```
Comment (pggl) [inline src/components/footer/Info.tsx]: remove this and you style instead

File: src/components/footer/Location.tsx [hunk 37]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import styled from "@emotion/styled";
+import Icon from "../Icon";
+import Place from "../../static/svg/icon-place.svg";
+
+const Street = styled.div`
+  width: 172px;
+  height: 36px;
+`;
```
Comment (pggl) [inline src/components/footer/Location.tsx]: remove this and you style instead

**Review by pggl — CHANGES_REQUESTED** [review]

**Pull request closed without merging**